contracts/erc721/PaymentObligation.sol

```diff
@@ -105,20 +105,67 @@ contract PaymentObligation is Initializable, UserMintableERC721 {
     uint256 _tokenId,
     string memory _tokenURI,
     uint256 _anchorId,
-    bytes32 _merkleRoot,
-    string[] memory _values,
+    uint256 _nextAnchorId,
+    bytes[] memory _properties,
+    bytes[] memory _values,
     bytes32[] memory _salts,
     bytes32[][] memory _proofs
   )
   public
   {
+    // Get the document root from AnchorRepository
+    bytes32 merkleRoot = super._getDocumentRoot(
+      _anchorId
+    );
+
+    // Enforce that there is not a newer version of the document on chain
+    super._isLatestDocumentVersion(
+      merkleRoot,
+      _nextAnchorId,
```

Comment:
The nextAnchorId is not in values because of the type. This way we avoid bytes to uint conversion in the contract. #savesomemoney

Reply:
We do not use all the properties and values(2 props out of 8,  4 values out of 8) from the proofs because most of them we reconstruct or hardcode in the contract and by having different length arrays we save some gas. This means we have stuff like:
```
      _properties[0],
      _values[3],
      _salts[5],
      _proofs[5]
```
Makes the api hard to understand because of the variable length arrays and the order. Any suggestions?